go-controller/pkg/util/node_annotations.go

```diff
@@ -685,3 +689,114 @@ func GetNodeZone(node *kapi.Node) string {
 
 	return zoneName
 }
+
+func parseNetworkIDsAnnotation(nodeAnnotations map[string]string, annotationName string) (map[string]string, error) {
+	annotation, ok := nodeAnnotations[annotationName]
+	if !ok {
+		return nil, newAnnotationNotSetError("could not find %q annotation", annotationName)
+	}
+
+	networkIdsStrMap := map[string]string{}
+	networkIds := make(map[string]string)
+	if err := json.Unmarshal([]byte(annotation), &networkIds); err != nil {
+		return nil, fmt.Errorf("could not parse %q annotation %q : %v",
+			annotationName, annotation, err)
+	}
+	for netName, v := range networkIds {
+		networkIdsStrMap[netName] = v
+	}
+
+	if len(networkIdsStrMap) == 0 {
+		return nil, fmt.Errorf("unexpected empty %s annotation", annotationName)
+	}
+
+	return networkIdsStrMap, nil
+}
+
+// ParseNetworkIDAnnotation parses the 'ovnNetworkIds' annotation for the specified
+// network in 'netName' and returns the network id.
+func ParseNetworkIDAnnotation(node *kapi.Node, netName string) (int, error) {
+	networkIdsMap, err := parseNetworkIDsAnnotation(node.Annotations, ovnNetworkIds)
+	if err != nil {
+		return -1, err
+	}
+
+	networkId, ok := networkIdsMap[netName]
+	if !ok {
+		return -1, newAnnotationNotSetError("node %q has no %q annotation for network %s", node.Name, ovnNetworkIds, netName)
+	}
+
+	return strconv.Atoi(networkId)
+}
+
+func updateNetworkIDsAnnotation(annotations map[string]string, netName string, netId int) error {
+	var bytes []byte
+
+	// First get the all network ids for all existing networks
+	networkIdsMap, err := parseNetworkIDsAnnotation(annotations, ovnNetworkIds)
+	if err != nil {
+		if !IsAnnotationNotSetError(err) {
+			return fmt.Errorf("failed to parse node network id annotation %q: %v",
+				annotations, err)
+		}
+		// in the case that the annotation does not exist
+		networkIdsMap = map[string]string{}
+	}
+
+	// add or delete network id of the specified network
+	if netId != -1 {
```

Comment:
need constants, I have no idea what -1 is

Reply:
Ack